[build]: Fix windows build after upgrading flux

After upgrading to flux v0.196.1 the build step for windows is failing due to linker errors:

```
/usr/bin/x86_64-w64-mingw32-ld: /root/.cache/go-build/pkgconfig/windows_amd64/lib/libflux-3eacf167e02a89d22ffdd59fd633a393685bd4797a5e9cca1dfcce74e0009bde.a(std-04636370035c201d.std.3b98da929d84a378-cgu.0.rcgu.o): in function `std::sys::windows::fs::open_link_no_reparse':
/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\std\src\sys\windows/fs.rs:829: undefined reference to `NtCreateFile'
/usr/bin/x86_64-w64-mingw32-ld: /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\std\src\sys\windows/fs.rs:858: undefined reference to `RtlNtStatusToDosError'
/usr/bin/x86_64-w64-mingw32-ld: /root/.cache/go-build/pkgconfig/windows_amd64/lib/libflux-3eacf167e02a89d22ffdd59fd633a393685bd4797a5e9cca1dfcce74e0009bde.a(std-04636370035c201d.std.3b98da929d84a378-cgu.0.rcgu.o): in function `std::sys::windows::c::NtReadFile':
/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\std\src\sys\windows/c.rs:283: undefined reference to `NtReadFile'
/usr/bin/x86_64-w64-mingw32-ld: /root/.cache/go-build/pkgconfig/windows_amd64/lib/libflux-3eacf167e02a89d22ffdd59fd633a393685bd4797a5e9cca1dfcce74e0009bde.a(std-04636370035c201d.std.3b98da929d84a378-cgu.0.rcgu.o): in function `std::sys::windows::handle::Handle::synchronous_read':
/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\std\src\sys\windows/handle.rs:273: undefined reference to `RtlNtStatusToDosError'
/usr/bin/x86_64-w64-mingw32-ld: /root/.cache/go-build/pkgconfig/windows_amd64/lib/libflux-3eacf167e02a89d22ffdd59fd633a393685bd4797a5e9cca1dfcce74e0009bde.a(std-04636370035c201d.std.3b98da929d84a378-cgu.0.rcgu.o): in function `std::sys::windows::c::NtWriteFile':
/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\std\src\sys\windows/c.rs:306: undefined reference to `NtWriteFile'
/usr/bin/x86_64-w64-mingw32-ld: /root/.cache/go-build/pkgconfig/windows_amd64/lib/libflux-3eacf167e02a89d22ffdd59fd633a393685bd4797a5e9cca1dfcce74e0009bde.a(std-04636370035c201d.std.3b98da929d84a378-cgu.0.rcgu.o): in function `std::sys::windows::handle::Handle::synchronous_write':
/rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\std\src\sys\windows/handle.rs:319: undefined reference to `RtlNtStatusToDosError'
collect2: error: ld returned 1 exit status
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -7,7 +7,7 @@ orbs:
 parameters:
   cross-container-tag:
     type: string
-    default: go1.23.5-latest
+    default: go1.23.5-ba67b11de54aa0bf8f1f92a81f786205de8024b6
 
   workflow:
     type: string

--- a/scripts/ci/build-tests.sh
+++ b/scripts/ci/build-tests.sh
@@ -29,6 +29,7 @@ function build_mac () {
 }
 
 function build_windows () {
+    CGO_LDFLAGS="-lntdll -ladvapi32 -lkernel32 -luserenv -lws2_32" \
     CGO_ENABLED=1 PKG_CONFIG=$(which pkg-config) CC="$(xcc windows)" go-test-compile \
         -tags sqlite_foreign_keys,sqlite_json,timetzdata -o "${1}/" -x ./...
 }
@@ -69,7 +70,6 @@ function main () {
             build_mac "$out_dir"
             ;;
         windows)
-            export CGO_LDFLAGS="-lntdll -ladvapi32 -lkernel32 -luserenv -lws2_32 ${CGO_LDFLAGS}"
             build_windows "$out_dir"
             ;;
         *)

--- a/scripts/ci/build-tests.sh
+++ b/scripts/ci/build-tests.sh
@@ -69,6 +69,7 @@ function main () {
             build_mac "$out_dir"
             ;;
         windows)
+            export CGO_LDFLAGS="-lntdll -ladvapi32 -lkernel32 -luserenv -lws2_32 ${CGO_LDFLAGS}"
             build_windows "$out_dir"
             ;;
         *)

--- a/scripts/ci/build.sh
+++ b/scripts/ci/build.sh
@@ -47,6 +47,7 @@ function main () {
                 "$pkg"
             ;;
         windows_amd64)
+            export CGO_LDFLAGS="-lntdll -ladvapi32 -lkernel32 -luserenv -lws2_32 ${CGO_LDFLAGS}"
             export CC="$(xcc windows)"
             CGO_ENABLED=1 PKG_CONFIG=$(which pkg-config) go build \
                 -tags assets,sqlite_foreign_keys,sqlite_json,timetzdata \

